.github/actions/setup-miniconda/action.yml

```diff
@@ -164,12 +164,24 @@ runs:
             AVAIL=$(echo $LINE | cut -f1 -d' ')
             MOUNT=$(echo $LINE | cut -f2 -d' ')
 
-            if [ "$MOUNT" = "/" ]; then
-              if [ "$AVAIL" -lt "$MINIMUM_AVAILABLE_SPACE_IN_KB" ]; then
-                echo "There is only ${AVAIL}KB free space left in $MOUNT, which is less than the minimum requirement of ${MINIMUM_AVAILABLE_SPACE_IN_KB}KB. Please help create an issue to PyTorch Release Engineering via https://github.com/pytorch/test-infra/issues and provide the link to the workflow run."
-                exit 1;
+            if [ "${MOUNT}" = "/" ]; then
+              if [ "${AVAIL}" -lt "${MINIMUM_AVAILABLE_SPACE_IN_KB}" ]; then
+                echo "There is only ${AVAIL}KB free space left in ${MOUNT}, which is less than the minimum requirement of ${MINIMUM_AVAILABLE_SPACE_IN_KB}KB for ${RUNNER_OS}"
+
+                if [ "${RUNNER_OS}" = "macOS" ]; then
+                  # We can clean up /System/Library/Caches/com.apple.coresymbolicationd on MacOS to free up the space and this should free up enough space
+                  # https://github.com/pytorch/pytorch/issues/85440
+                  sudo rm "/System/Library/Caches/com.apple.coresymbolicationd/data" || true
+                  # Stop the daemon and launchctl will automatically start it again, thus accomplish a restart and free up the above file
+                  sudo launchctl stop com.apple.coresymbolicationd
+
+                  df -h
+                else
+                  echo "Please help create an issue to PyTorch Release Engineering via https://github.com/pytorch/test-infra/issues and provide the link to the workflow run."
+                  exit 1;
+                fi
```

Comment:
Hmm, should we check if this achieve the desired effect and if not, still suggest that users should create an issue?

Reply:
Yeah, I'm thinking about this. On one hand, I'm quite confident that cleaning up the file would free up enough space. On the other hand, re-do the check does no harm but the code will need to be refactored a bit not turn into some kind of spaghetti code.  I should refactor this into a function